arch/arm64/include/spinlock.h

```diff
@@ -83,6 +83,25 @@
  * -- Clear Exclusive access monitor (CLREX) This is used to
  *     clear the state of the Local Exclusive Monitor.
  */
+#if defined(CONFIG_TICKET_SPINLOCK)
+#include <stdatomic.h>
+
+/* Memory layout is related uint64_t in arm64 little endian
+ * if uint64 is 0x010203040A0B0C0D, the next is 0A0B0C0D and
+ * the owner is 01020304.
+ */
+
+struct ticket_spinlock_s
+{
+    atomic_uint next;
```

Comment:
it's better to use 16bit count on 32bit machine

Reply:
You are right but the arch is arm64.
If the machine is 32bit, I will use 16bit counter.

Reply:
@xiaoxiang781216 Sorry but I have a question is is it possible use 32bit spinlock on 64bit machine?
I found the [page.](http://lists.infradead.org/pipermail/linux-arm-kernel/2013-October/203491.html)
In the bottom of page, it shows ticket spinlock is 32bit even on 64 bit machine.

I've attempted to modify spinlock_t to 32bit but it failed.

Reply:
Yes, atomic operation can be done on the size smaller than native size(4B for 32bit, 8B for 64bit).